src/components/VideoPlayerContexts/PlaybackContext.tsx

```diff
@@ -73,21 +76,22 @@ function PlaybackContextProvider({children}: ChildrenProps) {
     );
 
     const resetVideoPlayerData = useCallback(() => {
-        videoResumeTryNumberRef.current = 0;
-        stopVideo();
-        setCurrentlyPlayingURL(null);
-        setSharedElement(null);
-        setOriginalParent(null);
-        currentVideoPlayerRef.current = null;
-        unloadVideo();
+        (playVideoPromiseRef.current ?? Promise.resolve()).then(stopVideo).finally(() => {
+            videoResumeTryNumberRef.current = 0;
+            setCurrentlyPlayingURL(null);
+            setSharedElement(null);
+            setOriginalParent(null);
+            currentVideoPlayerRef.current = null;
+            unloadVideo();
+        });
     }, [stopVideo, unloadVideo]);
 
     useEffect(() => {
-        if (!currentReportID) {
+        if (!currentReportID || !prevCurrentReportID || currentReportID === '-1' || prevCurrentReportID === '-1') {
```

Comment:
Yes. The first one waits until the play promise is completed. But on Android mWeb (if I remember the platform correctly), the play is never triggered anymore after the video is stopped. So,

1. play is called
2. resetVideoPlayer is called because currentReportID changes from an empty string to something
3. There is no play called anymore like in other platforms, so the video is completely stopped and (ref) cleared.

Reply:
Nope, what I mean is that, when we open the video, the play video is called multiple times.
https://github.com/Expensify/App/blob/8193c379328c33519586e025262c62f63f37b882/src/components/VideoPlayerContexts/PlaybackContext.tsx#L28-L36

And then `resetVideoPlayerData` is also called once. On other platforms, the order of call looks like this:
<img width="143" alt="Screenshot 2024-08-07 at 22 19 53" src="https://github.com/user-attachments/assets/f0452554-3f18-4f75-93d0-bb5e93f9bd12">

In mWeb Android, it's
play
play
...
reset
[no more call]

But I just retested it and the call order is similar to other platforms too. 

> we only need to check !currentReportID || currentReportID === '-1' is enough, what do you think?

Still, we need to prevent the reset to be called, so we need to check for `prevCurrentReportID` too.

> And do you think we should make -1 as a constant?

I don't have a strong opinion about this one. There is -1 and '-1' and we use this everywhere on the app without const. Maybe we can have a new cleanup issue to create the const and replace it everywhere in the app.

Reply:
Added comment